app/routers/calendar_grid.py

```diff
@@ -0,0 +1,197 @@
+import calendar
+import itertools
+import locale
+from datetime import date, datetime, timedelta
+from typing import Any, Iterator, List
+
+DISPLAY_BLOCK = 100
+MONTH_BLOCK = 6
+
+locale.setlocale(locale.LC_TIME, ("en", "UTF-8"))
+
+
+class Day:
+    """A Day class.
+
+    Args:
+        date    (datetime): A single datetime date.
+    Arguments:
+        date    (datetime): A single datetime date.
+        sday         (str): String format 00 of the day number.
+        dailyevents (List): List of tuples represent daily event information.
+        events      (List): List of tuples represent time event name.
+                            EX: [("09AP", "Meeting with yam")]
+        css         (Dict): All css classes represent day.
+    """
+
+    def __init__(self, date: datetime):
+        self.date = date
+        self.sday = self.date.strftime("%A")
+        self.dailyevents = []
+        self.events = []
+        self.css = {
```

Comment:
It is usually a very bad idea to mix the view and the code layers. If possible, separate them.

Reply:
The idea was to create a dynamic "day" graphic properties that can be saved and changes as u can see in the object it self there is all day elements and then only class names according to the css setup. (this way u can change and control all every day + its sub elements. in the future maybe even as a "Theme Class". 

I need to investigate how it can be achieve in different approaches.
Ill start by reading the link here.